Vienna/Sources/Main window/FoldersTree.m

```diff
@@ -1634,5 +1569,32 @@ - (NSString *)outlineView:(NSOutlineView *)outlineView toolTipForCell:(NSCell *)
     return nil;
 }
 
+// MARK: - NSTextFieldDelegate
+
+// Called when the user finishes editing a folder name
+- (void)controlTextDidEndEditing:(NSNotification *)obj {
+    TreeNode * node = (TreeNode *)[self.outlineView itemAtRow:self.outlineView.selectedRow];
+    NSTextField *textField = (NSTextField *)obj.object;
+    NSString * newName = textField.stringValue;
+    Folder * folder = node.folder;
+    
+    // Remove the "☁️ " symbols on Open Reader feeds
+    if (folder.type == VNAFolderTypeOpenReader && [newName hasPrefix:@"☁️ "]) {
+        NSString *tmpName = [newName substringFromIndex:3];
+        newName = tmpName;
+    }
+    
+    if (![folder.name isEqualToString:newName]) {
+        Database * dbManager = [Database sharedManager];
+        if ([dbManager folderFromName:newName] != nil) {
+            runOKAlertPanel(NSLocalizedString(@"Cannot rename folder", nil), NSLocalizedString(@"A folder with that name already exists", nil));
+        } else {
+            [dbManager setName:newName forFolder:folder.itemId];
+            if (folder.type == VNAFolderTypeOpenReader) {
+                [[OpenReader sharedManager] setFolderTitle:newName forFeed:folder.remoteId];
+            }
+        }
+    }
```

Comment:
Not sure if that´s possible, but since you asked for feedback: This part feels like it should be in the setter or a set method of `folder.name` instead of an action callback. That way we are able to set the folder name from other places, too.

Reply:
Yeah I don't like it this way either. I was thinking I would search through the code later and see if `folder.name` is ever set without being persisted in the database.

Reply:
@TAKeanice I'll leave this as is for now since it is just a cut and paste of existing code. It's obviously far from ideal but we need to re-implement our whole model which should be done in a separate PR.